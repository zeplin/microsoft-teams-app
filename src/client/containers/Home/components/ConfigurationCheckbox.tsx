import React, { FunctionComponent } from "react";
import { Checkbox, Flex, Text } from "@fluentui/react-northstar";

interface ConfigurationCheckbox {
    title: string;
    description: string;
}

export const ConfigurationCheckbox: FunctionComponent<ConfigurationCheckbox> = ({
    title,
    description
}) => (
    <div>
        <Checkbox defaultChecked label={
            <Flex fill column>
                <Text>{title}</Text>
                <Text size="small">{description}</Text>
            </Flex>
        } />
    </div>
);
