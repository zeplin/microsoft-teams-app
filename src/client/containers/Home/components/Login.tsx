import React, { FunctionComponent } from "react";
import { Button, Flex, Text } from "@fluentui/react-northstar";
import * as microsoftTeams from "@microsoft/teams-js";
import { BASE_URL } from "../../../config";

interface LoginProps {
    onTokenReceive: (token: string) => void;
}

export const Login: FunctionComponent<LoginProps> = ({ onTokenReceive }) => (
    <Flex fill column gap="gap.small" styles={{ padding: "60px" }}>
        <Text size="large" content="One more step before finishing connection" weight="semibold" />
        <Flex fill column gap="gap.medium">
            <Text>
                Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once
                itâ€™s authorized, you can do more advanced settings on the Teams channel linked with Zeplin.
            </Text>
            <Button
                onClick={(): void => {
                    microsoftTeams.authentication.authenticate({
                        successCallback: onTokenReceive,
                        url: `${BASE_URL}/zeplin/auth/start`
                    });
                }}>
                Log in Zeplin
            </Button>
        </Flex>
    </Flex>
);
