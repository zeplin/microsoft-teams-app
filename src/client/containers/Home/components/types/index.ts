export * from "./Workspace";
export * from "./WorkspaceType";
export * from "./Resource";
export * from "./ResourceType";
export * from "./WebhookEvent";
