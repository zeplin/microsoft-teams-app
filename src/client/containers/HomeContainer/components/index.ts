export * from "./Login";
export * from "./Configuration";
