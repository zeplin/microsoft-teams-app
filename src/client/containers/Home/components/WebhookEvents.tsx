import React, { FunctionComponent } from "react";
import { ResourceType, WebhookEvent } from "./types";
import { Flex } from "@fluentui/react-northstar";
import { ConfigurationCheckbox } from "./ConfigurationCheckbox";

const HALF_DIVIDER = 2;

const webhookEvents: WebhookEvent[] = [
    {
        title: "Screens",
        description: "Screens added",
        resourceTypes: [ResourceType.PROJECT]
    },
    {
        title: "Screen versions",
        description: "Versions added to screen",
        resourceTypes: [ResourceType.PROJECT]
    },
    {
        title: "Components",
        description: "Components added, removed or updated on Styleguide",
        resourceTypes: [ResourceType.PROJECT, ResourceType.STYLEGUIDE]
    },
    {
        title: "Colors",
        description: "Colors added, removed or updated on Styleguide",
        resourceTypes: [ResourceType.PROJECT, ResourceType.STYLEGUIDE]
    },
    {
        title: "Text styles",
        description: "Text styles added, removed or updated on Styleguide",
        resourceTypes: [ResourceType.PROJECT, ResourceType.STYLEGUIDE]
    },
    {
        title: "Spacing tokens",
        description: "Spacing tokens added, removed or updated on Styleguide",
        resourceTypes: [ResourceType.PROJECT, ResourceType.STYLEGUIDE]
    },
    {
        title: "Notes",
        description: "Note added",
        resourceTypes: [ResourceType.PROJECT]
    },
    {
        title: "Replies",
        description: "Replied to note",
        resourceTypes: [ResourceType.PROJECT]
    },
    {
        title: "Members",
        description: "Members invited",
        resourceTypes: [ResourceType.PROJECT, ResourceType.STYLEGUIDE]
    }
];

interface WebhookEventsProps {
    resourceType: ResourceType;
}

export const WebhookEvents: FunctionComponent<WebhookEventsProps> = ({
    resourceType
}) => {
    const availableEvents = webhookEvents.filter(({ resourceTypes }) => resourceTypes.includes(resourceType));
    const middleIndex = Math.ceil(availableEvents.length / HALF_DIVIDER);
    return (
        <Flex fill gap="gap.small">
            <Flex fill column gap="gap.smaller">
                {
                    availableEvents.slice(0, middleIndex).map(({ title, description }) => (
                        <ConfigurationCheckbox key={title} title={title} description={description} />
                    ))
                }
            </Flex>
            <Flex fill column gap="gap.smaller">
                {
                    availableEvents.slice(middleIndex).map(({ title, description }) => (
                        <ConfigurationCheckbox key={title} title={title} description={description} />
                    ))
                }
            </Flex>
        </Flex>
    );
};
