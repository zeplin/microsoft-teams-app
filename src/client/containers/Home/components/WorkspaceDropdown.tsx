import React, { FunctionComponent } from "react";
import { Dropdown } from "@fluentui/react-northstar";

import { Workspace, WorkspaceType } from "./types";

interface WorkspaceDropdownProps {
    loading: boolean;
    organizations: {
        name: string;
        id: string;
    }[];
    onChange: (value: Workspace | undefined) => void;
}

export const WorkspaceDropdown: FunctionComponent<WorkspaceDropdownProps> = ({
    loading,
    organizations,
    onChange
}) => <Dropdown
    loading={loading}
    fluid
    checkable
    items={[
        {
            header: "Personal Workspace",
            onClick: (): void => onChange({ type: WorkspaceType.PERSONAL })
        },
        ...organizations.map(({ name, id }) => ({
            header: `${name}'s Workspace`,
            onClick: (): void => onChange({ type: WorkspaceType.ORGANIZATION, organizationId: id })
        }))
    ]}
    placeholder="Select Workspace"
/>;
