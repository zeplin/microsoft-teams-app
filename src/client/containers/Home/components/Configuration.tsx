import React, { FunctionComponent, useState } from "react";
import { Flex, Text } from "@fluentui/react-northstar";
import Axios from "axios";
import { useQuery } from "react-query";
import { useRouter } from "next/router";

import { BASE_URL } from "../../../config";
import { WorkspaceDropdown } from "./WorkspaceDropdown";
import { Resource, ResourceType, Workspace } from "./types";
import { ResourceDropdown } from "./ResourceDropdown";
import { WebhookEvents } from "./WebhookEvents";

interface ConfigurationProps {
    accessToken: string;
}

export const Configuration: FunctionComponent<ConfigurationProps> = ({
    accessToken
}) => {
    const {
        query: {
            channel
        }
    } = useRouter();

    const { isLoading: isOrganizationsLoading, data: organizations } = useQuery(
        "organizations",
        async () => {
            const { data: result } = await Axios.get(
                `${BASE_URL}/api/organizations`,
                {
                    headers: {
                        Authorization: `Bearer ${accessToken}`
                    }
                }
            );
            return result;
        },
        {
            initialData: []
        }
    );

    const [selectedWorkspace, setSelectedWorkspace] = useState<Workspace>();
    const [selectedResource, setSelectedResource] = useState<Resource | undefined>();

    return (
        <Flex fill column gap="gap.large">
            <div />
            <Flex fill column gap="gap.small">
                <Text weight="semibold">
                    You are connecting <Text weight="bold">#{channel}</Text> to Zeplin.
                </Text>
                <Text>
                    Can{"'"}t find the project/styleguide to connect to?
                    {" "}
                    <Text
                        as="a"
                        color="brand"
                        href="https://zpl.io/msteams-app-docs"
                        target="_blank"
                        styles={{
                            "textDecoration": "none",
                            ":hover": {
                                textDecoration: "underline"
                            }
                        }}>
                        Learn more about permissions.
                    </Text>
                </Text>
                <Flex fill gap="gap.small">
                    <Flex.Item grow shrink={0} styles={{ flexBasis: 0 }}>
                        <div>
                            <WorkspaceDropdown
                                loading={isOrganizationsLoading}
                                organizations={organizations}
                                onChange={(workspace): void => {
                                    setSelectedWorkspace(workspace);
                                    setSelectedResource(undefined);
                                }} />
                        </div>
                    </Flex.Item>
                    <Flex.Item grow shrink={0} styles={{ flexBasis: 0 }}>
                        <div>
                            <ResourceDropdown
                                disabled={!selectedWorkspace}
                                onChange={(resource): void => setSelectedResource(resource)} />
                        </div>
                    </Flex.Item>
                </Flex>
            </Flex>
            <Flex fill column gap="gap.medium">
                <Text weight="semibold">
                    Select the events you want to get a message for:
                </Text>
                <WebhookEvents resourceType={selectedResource?.type ?? ResourceType.PROJECT} />
            </Flex>
        </Flex>
    );
};
