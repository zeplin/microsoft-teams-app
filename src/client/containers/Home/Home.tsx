import React, {
    FunctionComponent, Reducer, useEffect, useReducer
} from "react";
import * as microsoftTeams from "@microsoft/teams-js";
import { Loader } from "@fluentui/react-northstar";
import { Login } from "./components";

enum Status {
    LOADING,
    LOGIN,
    CONFIGURATION
}

enum ActionType {
    COMPLETE_LOADING,
    GET_TOKEN
}

type State = {
    status: Status.LOADING | Status.LOGIN;
} | {
    status: Status.CONFIGURATION;
    accessToken: string;
}

type Action = {
    type: ActionType.COMPLETE_LOADING;
} | {
    type: ActionType.GET_TOKEN;
    value: string;
}

const reducer: Reducer<State, Action> = (state, action) => {
    switch (action.type) {
        case ActionType.COMPLETE_LOADING:
            return { status: Status.LOGIN };
        case ActionType.GET_TOKEN:
            return { status: Status.CONFIGURATION, accessToken: action.value };
        default:
            throw new Error();
    }
};

export const Home: FunctionComponent = () => {
    const [state, dispatch] = useReducer(reducer, { status: Status.LOADING });

    useEffect(() => {
        microsoftTeams.initialize(() => {
            microsoftTeams.appInitialization.notifySuccess();
            dispatch({ type: ActionType.COMPLETE_LOADING });
        });
    }, []);

    switch (state.status) {
        case Status.LOADING:
            return <Loader id="loader-home"/>;
        case Status.LOGIN:
            return <Login onTokenReceive={(value): void => dispatch({ type: ActionType.GET_TOKEN, value })} />;
        case Status.CONFIGURATION:
            return (
                <div>
                    {state.accessToken}
                </div>
            );
        default:
            throw new Error();
    }
};
