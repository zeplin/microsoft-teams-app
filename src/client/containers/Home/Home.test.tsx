import React from "react";
import { Home } from "./Home";
import * as microsoftTeams from "@microsoft/teams-js";
import { fireEvent, render, RenderResult } from "@testing-library/react";
import { Providers } from "../../components";

jest.mock("@microsoft/teams-js", () => ({
    initialize: jest.fn(callback => callback()),
    appInitialization: {
        notifySuccess: jest.fn()
    },
    settings: {
        registerOnSaveHandler: jest.fn(),
        registerOnRemoveHandler: jest.fn(),
        setValidityState: jest.fn(),
        setSettings: jest.fn(),
        getSettings: jest.fn(callback => callback({ entityId: "" }))
    },
    authentication: {
        authenticate: jest.fn(({ successCallback }) => successCallback("accessToken"))
    }
}));

jest.mock("../../config", () => ({
    BASE_URL: "https://test.com"
}));

function renderHome(): RenderResult {
    return render(
        <Providers>
            <Home />
        </Providers>
    );
}

describe("Home", () => {
    it("should render initially", () => {
        const spy = jest.spyOn(microsoftTeams, "initialize")
            .mockImplementation(() => undefined);
        const { container: { firstChild: component } } = renderHome();
        expect(component).toMatchSnapshot();
        spy.mockImplementation(callback => callback());
    });

    it("should initialize Microsoft Teams", () => {
        const spy = jest.spyOn(microsoftTeams.appInitialization, "notifySuccess");

        const { container: { firstChild: component } } = renderHome();

        expect(component).toMatchSnapshot();
        expect(spy).toBeCalledWith();
    });

    it("should change ui when login flow completed", () => {
        const spy = jest.spyOn(microsoftTeams.appInitialization, "notifySuccess");

        const { container: { firstChild: component }, getByText } = renderHome();

        fireEvent.click(getByText("Log in Zeplin"));

        expect(component).toMatchSnapshot();
        expect(spy).toBeCalledWith();
    });
});
