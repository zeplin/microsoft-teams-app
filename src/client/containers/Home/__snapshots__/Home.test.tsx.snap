// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should change ui when login flow completed 1`] = `
<div
  class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-flex am ao bq br bs"
    data-uses-unhanded-props="true"
  >
    <div />
    <div
      class="ui-flex am ao bq br fq"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text bu"
        data-uses-unhanded-props="true"
      >
        You are connecting 
        <span
          class="ui-text fr"
          data-uses-unhanded-props="true"
        >
          #
          channel
        </span>
         to Zeplin.
      </span>
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
      >
        Can
        '
        t find the project/styleguide to connect to?
         
        <a
          class="ui-text fs ft fu"
          data-uses-unhanded-props="true"
          dir="auto"
          href="https://zpl.io/msteams-app-docs"
          target="_blank"
        >
          Learn more about permissions.
        </a>
      </span>
      <div
        class="ui-flex am bq br fv"
        data-uses-unhanded-props="true"
      >
        <div
          class="ui-dropdown"
        >
          <div
            class="ui-dropdown__container am fw bb cp cq cr cs fx fy fz ga ck bq gb gc gd ge gf gg gh cz gi gj gk gl gm gn"
          >
            <div
              class="ui-dropdown__selected-items am fw go gp bq gq"
            >
              <button
                aria-haspopup="true"
                aria-labelledby="dropdown-trigger-button-1"
                class="ui-button bx by gr b gs cb cc gt al bb cd gu cf gv ch ci cj ck cl cm cn co cp cq cr cs fx fy fz ga gw gx gy gi gj gk gl de gz ha hb hc hd he dl hf hg dm hh hi hj hk hl hm hn ho hp hq hr hs ht hu hv hw hx hy hz ia ib ic id ie if ig ih ii ij ik dn do dp dq dr ds dt du dv dw dx dy il im in io ed ip ef eg eh ei ej ek el em en eo ep eq er es et iq ir is it ey iu fb fc fd iv iw fe ff fg fh fi fj fk fl bq ar ix iy iz ja ui-dropdown__trigger-button"
                data-aa-class="Button"
                data-toggle="true"
                data-uses-unhanded-props="true"
                id="dropdown-trigger-button-1"
                role="button"
                type="button"
              >
                <span
                  class="ui-button__content ar fm fn fo bu fp"
                  data-uses-unhanded-props="true"
                  dir="auto"
                >
                  Select Workspace
                </span>
              </button>
            </div>
            <div
              aria-hidden="true"
              class="ui-box al am an jb jc jd ci je ix iy iz ja jf jg br jh ui-dropdown__toggle-indicator"
              data-aa-class="Box"
              data-uses-unhanded-props="true"
              role="img"
            />
            <ul
              aria-hidden="true"
              aria-labelledby="downshift-0-label"
              class="ui-list ba ji cd jj cf jk ix iy iz ja ck cp cq cr cs gb gc gd ge fx fy fz ga jl jm go bq a jn ui-dropdown__items-list"
              data-aa-class="List"
              data-uses-unhanded-props="true"
              id="downshift-0-menu"
              role="listbox"
              tabindex="-1"
            />
          </div>
        </div>
        <div
          class="ui-dropdown"
        >
          <div
            class="ui-dropdown__container am fw bb cp cq cr cs jo jp jq jr ck bq gb gc gd ge gf js gh cz jt gn je ju"
          >
            <div
              class="ui-dropdown__selected-items am fw go gp bq gq"
            >
              <button
                aria-disabled="true"
                aria-haspopup="true"
                aria-labelledby="dropdown-trigger-button-2"
                class="ui-button bx by gr jv gs cb cc gt al bb cd gu cf gv ch jw cj ck cl cm cn co cp cq cr cs fx fy fz ga gw gx gy gi gj gk gl de gz ha hb hc hd he dl hf hg dm hh hi hj hk hl hm hn ho hp hq hr hs ht hu hv hw hx hy hz ia ib ic id ie if ig ih ii ij ik dn do dp dq dr ds dt du dv dw dx dy il im in io ed ip ef eg eh ei ej ek el em en eo ep eq er es et iq ir is it ey iu fb fc fd iv iw fe ff fg fh fi fj fk fl jx bq ar ix iy iz ja ui-dropdown__trigger-button"
                data-aa-class="Button"
                data-toggle="true"
                data-uses-unhanded-props="true"
                disabled=""
                id="dropdown-trigger-button-2"
                role="button"
                type="button"
              >
                <span
                  class="ui-button__content ar fm fn fo bu fp"
                  data-uses-unhanded-props="true"
                  dir="auto"
                >
                  Select Project/Styleguide
                </span>
              </button>
            </div>
            <div
              aria-hidden="true"
              class="ui-box al am an jy jc jd jw je ix iy iz ja jf jg br jh ui-dropdown__toggle-indicator"
              data-aa-class="Box"
              data-uses-unhanded-props="true"
              role="img"
            />
            <ul
              aria-hidden="true"
              aria-labelledby="downshift-1-label"
              class="ui-list ba ji cd jj cf jk ix iy iz ja ck cp cq cr cs gb gc gd ge fx fy fz ga jl jm go bq a jn ui-dropdown__items-list"
              data-aa-class="List"
              data-uses-unhanded-props="true"
              id="downshift-1-menu"
              role="listbox"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui-flex am ao bq br bt"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text bu"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        Select the events you want to get a message for:
      </span>
      <div
        class="ui-flex am bq br fv"
        data-uses-unhanded-props="true"
      >
        <div
          class="ui-flex am ao bq br jz"
          data-uses-unhanded-props="true"
        >
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Screens
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Screens added
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Screen versions
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Versions added to screen
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Components
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Components added, removed or updated on Styleguide
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Colors
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Colors added, removed or updated on Styleguide
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Text styles
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Text styles added, removed or updated on Styleguide
                  </span>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ui-flex am ao bq br jz"
          data-uses-unhanded-props="true"
        >
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Spacing tokens
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Spacing tokens added, removed or updated on Styleguide
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Notes
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Note added
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Replies
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Replied to note
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <div
              aria-checked="true"
              class="ui-checkbox bb ka kb kc ci ck b kd ke kf kg ch kh dm ih ii ij ik dn do dp dq dr ds dt ki kj kk kl dy dz ea eb ec ed ee ef eg km ei ej ek el em en eo kn ko kp kq et eu ev ew ex ey ez fb fc kr cy ks kt ku kv kw"
              data-aa-class="Checkbox"
              data-is-focusable="true"
              data-uses-unhanded-props="true"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="ui-box kx ky kz la lb lc ld le lf lg cp cq cr cs gh lh li lj lk ll lm iy iz ja ln lo lp lq je lr jc jd ui-checkbox__indicator"
                data-uses-unhanded-props="true"
              />
              <span
                class="ui-text ba ls lt ui-checkbox__label"
                data-uses-unhanded-props="true"
              >
                <div
                  class="ui-flex am ao bq br"
                  data-uses-unhanded-props="true"
                >
                  <span
                    class="ui-text"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Members
                  </span>
                  <span
                    class="ui-text lu bw"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Members invited
                  </span>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home should initialize Microsoft Teams 1`] = `
<div
  class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-flex am ao bq br bs"
    data-uses-unhanded-props="true"
  >
    <div />
    <div
      class="ui-flex am ao bq br bt"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text bu bv bw"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        One more step before finishing connection
      </span>
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once it’s authorized, you can do more advanced settings on the Teams channel linked with Zeplin.
      </span>
    </div>
    <div>
      <button
        class="ui-button bx by bz b ca cb cc an al bb cd ce cf cg ch ci cj ck cl cm cn co cp cq cr cs ct cu cv cw cx cy cz da db dc dd de df dg dh di dj dk dl dm dn do dp dq dr ds dt du dv dw dx dy dz ea eb ec ed ee ef eg eh ei ej ek el em en eo ep eq er es et eu ev ew ex ey ez fb fc fd fe ff fg fh fi fj fk fl"
        data-aa-class="Button"
        data-uses-unhanded-props="true"
      >
        <span
          class="ui-button__content ar fm fn fo bu fp"
          data-uses-unhanded-props="true"
          dir="auto"
        >
          Log in Zeplin
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Home should render initially 1`] = `
<div
  class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-loader al am an ao"
    data-aa-class="Loader"
    data-uses-unhanded-props="true"
    id="loader-home"
    role="progressbar"
  >
    <div
      class="ui-box ap aq ar ui-loader__indicator"
      data-uses-unhanded-props="true"
    >
      <div
        class="ui-box as at au av aw ax ay az ba ar bb bc bd be bf bg bh bi bj bk bl bm bn bo bp ui-loader__svg"
        data-uses-unhanded-props="true"
      />
    </div>
  </div>
</div>
`;
